src/main/java/duke/dao/TaskDaoImpl.java

```diff
@@ -0,0 +1,89 @@
+package duke.dao;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class TaskDaoImpl implements TaskDao {
+    private static Path taskDataPath = Paths.get(System.getProperty("user.dir"), "data", "task.txt");
+    private static File taskFile = new File(String.valueOf(taskDataPath));
+
+    public TaskDaoImpl() {
+        try {
+            taskFile.getParentFile().mkdirs();
+            taskFile.createNewFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public ArrayList<Task> loadAllTasks() {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try {
+            Scanner sc = new Scanner(taskFile);
+            while (sc.hasNext()) {
+                String line = sc.nextLine();
+                char taskType = line.charAt(1);
+                boolean isDone = (String.valueOf(line.charAt(4)).compareTo("\u2718") == 0) ? true : false;

```

Comment:
You might want to declare the unicode character as a constant with a descriptive name